src/Compiler/Checking/PostInferenceChecks.fs

```diff
@@ -383,7 +397,24 @@ let rec CheckTypeDeep (cenv: cenv) (visitTy, visitTyconRefOpt, visitAppTyOpt, vi
         CheckTypeDeep cenv f g env typeInstParentOpt body
         tps |> List.iter (fun tp -> tp.Constraints |> List.iter (CheckTypeConstraintDeep cenv f g env))
 
-    | TType_measure _ -> ()
+    | TType_measure tm ->
+        let checkAttribs tm m =
+            match tm with
+            | MeasureAttrib attribs when not attribs.IsEmpty -> CheckFSharpAttributes g attribs m |> CommitOperationResult
+            | _ -> ()
+
+        match tm with
```

Comment:
TypedTreeOps should already have helper functions for flattening units of measure types, couldn't `ListMeasureVarOccs ` be used (and the list then fed into checkAttribs) ?

Reply:
`ListMeasureVarOccs` Did'nt work but it seems that  `ListMeasureConOccsWithNonZeroExponents` does what we need